dogfood/Dockerfile.nix

```diff
@@ -0,0 +1,49 @@
+# Build stage
+FROM nixos/nix:2.19.2 as nix
+
+# enable --experimental-features 'nix-command flakes' globally
+RUN mkdir -p /etc/nix && echo "experimental-features = nix-command flakes" >> /etc/nix/nix.conf
+
+# Copy the Nix related files into the Docker image
+COPY flake.nix /app/flake.nix
+COPY flake.lock /app/flake.lock
+
+# Install dependencies from flake and remove the flake
+RUN nix profile install /app#all --priority 4 && rm -rf /app
```

Comment:
I can see this getting good mileage out of the Depot cache

Reply:
Might be a good idea to quote "/app#all" just to make it very clear it's as intended (bonus, it'll fix GitHub syntax highlighting 😂)